### ISSUE TITLE
javalib: `java.nio.file.Files.copy(Path, Path)` should handle symbolic links

### ISSUE THREAD

**Issue by WojciechMazur** [scala-native/scala-native#3912]
Currently trying to copy symbolic link will result in `IOException(s"Unsupported operation: copy symbolic link...")` 
Javadoc stats that: `If symbolic links are supported, and the file is a symbolic link, then the final target of the link is copied.`  

Also under `NOFOLLOW_LINKS`: 
```
Symbolic links are not followed. If the file is a symbolic link, then the symbolic link itself, not the target of the link, is copied. 
It is implementation specific if file attributes can be copied to the new link. 
In other words, the COPY_ATTRIBUTES option may be ignored when copying a symbolic link.
``` 

Blocker for https://github.com/com-lihaoyi/os-lib/pull/257
